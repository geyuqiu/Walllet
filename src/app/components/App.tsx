import React from "react";
import {HashRouter} from "react-router-dom";

import {Routing} from "../router/Routing";
import {Footer} from "./Footer/Footer";

const notProd = process.env.NODE_ENV !== "production";

export const App = () => {
	const mainClassName = "bg-white min-h-full";
	const showDebugScreensOnDev: string = notProd
		? `debug-screens ${mainClassName}`
		: mainClassName;

	return (
		<HashRouter>
			{/*<Header />*/}
			<main className={showDebugScreensOnDev} role="main">
				<Routing />
			</main>
			<Footer />
		</HashRouter>
	);
};
