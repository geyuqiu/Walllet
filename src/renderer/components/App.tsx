import React from "react";
import { HashRouter } from "react-router-dom";

import { Routing } from "../router/Routing";
import { Footer } from "./Footer/Footer";
import { Header } from "./Header/Header";

const notProd = process.env.NODE_ENV !== "production";

export const App = () => {
	const showDebugScreensOnDev: string = notProd
		? "debug-screens bg-white min-h-70vh"
		: "";

	return (
		<HashRouter>
			<Header />
			<main className={showDebugScreensOnDev} role="main">
				<Routing />
			</main>
			<Footer />
		</HashRouter>
	);
};
