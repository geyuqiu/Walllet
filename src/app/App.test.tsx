import {render, screen} from "@testing-library/react";
import React from "react";
import {MemoryRouter} from "react-router-dom";

import {App} from "./App";

describe("App", () => {
	it("should match snapshot", () => {
		const { container } = render(
			<MemoryRouter>
				<App />
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});

	it("should show debug in TEST ENV", () => {
		render(
			<MemoryRouter>
				<App />
			</MemoryRouter>
		);
		expect(screen.getByRole("main")).toHaveClass("debug-screens");
	});
});
