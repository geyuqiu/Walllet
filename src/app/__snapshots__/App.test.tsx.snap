// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div>
  <main
    class="debug-screens bg-white min-h-full"
    role="main"
  >
    <div
      class="Cardstyles__Wrapper-xnuhyv-0 fNkVcc bg-green-dark sm:flex sm:justify-center py-12 px-8"
      data-testid="Card"
    >
      <div
        class="Cardstyles__Wrapper-xnuhyv-0 fNkVcc bg-black-darkest p-8 justify-between text-white rounded-lg w-full max-w-screen-xl flex justify-center lg:items-center flex-col lg:flex-row"
        data-testid="Card"
      >
        <div
          class="flex items-center my-auto cursor-pointer"
        >
          <div
            class="sc-gsTCUz dXshFa"
            width="57"
          >
            <svg
              width="44"
            >
              logo.svg
            </svg>
          </div>
          <span
            class="text-2xl font-bold lg:hidden xl:block"
            data-testid="logo__text"
          >
            ARK Wallet
          </span>
        </div>
        <div
          class="lg:hidden"
        >
          <div
            class="Divider-sc-10xryrv-0 ijsPwK border-black-dark dark:border-theme-secondary-600"
            type="horizontal"
          />
        </div>
        <div
          class="lg:contents hidden"
        >
          <div
            class="Divider-sc-10xryrv-0 dCquGe border-black-dark dark:border-theme-secondary-600"
            type="vertical"
          />
        </div>
        <div
          class="relative border border-gray-darkest border-opacity-10 rounded-l-3xl rounded-r-3xl"
        >
          <span
            data-testid="dropdown__toggle"
          >
            <button
              class="flex justify-between items-center h-11 w-full focus:outline-none"
            >
              <div
                class="flex items-center"
              >
                <div
                  class="hidden sm:block"
                >
                  <div
                    class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy rounded-l-3xl border-r bg-black-light ring-theme-background"
                  >
                    <div
                      class="sc-bdfBwQ dfWOLJ ml-2"
                      data-testid="dropdown__left__icon"
                      height="15"
                      width="15"
                    >
                      <svg
                        fill="#000000"
                        stroke="#FBC457"
                      >
                        address.svg
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  class="ml-5"
                >
                  <span
                    class="hidden sm:block lg:hidden xl:block"
                  />
                  <span
                    class="block sm:hidden"
                  />
                  <span
                    class="hidden lg:block xl:hidden"
                  />
                </div>
              </div>
              <div
                class="sc-bdfBwQ dfWOLJ transition-transform mx-5"
                data-testid="dropdown__right_icon"
                height="15"
                width="15"
              >
                <svg>
                  arrow-down.svg
                </svg>
              </div>
            </button>
          </span>
        </div>
        <div
          class="hidden lg:contents"
        >
          <div
            class="Divider-sc-10xryrv-0 dCquGe border-black-dark dark:border-theme-secondary-600"
            type="vertical"
          />
        </div>
        <div
          class="flex flex-row mt-6 lg:mt-0"
        >
          <div
            class="hidden sm:block"
          >
            <div
              class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy border rounded-full mr-3 ring-theme-background"
            >
              <div
                class="sc-bdfBwQ dfWOLJ"
                height="15"
                width="15"
              >
                <svg
                  fill="#000000"
                  stroke="#FBC457"
                >
                  balance.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col"
          >
            <span
              class="text-gray-darkest"
            >
              Balance
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ml-6 mr-3 sm:mx-8 sm:flex sm:justify-center pt-12"
    >
      <p
        data-testid="TransactionTable__Loading"
      >
        Loading Transactions ...
      </p>
    </div>
  </main>
  <footer
    class="bg-gray-light flex items-center justify-center p-4 fixed bottom-0 w-full z-100 text-black-lightest text-sm"
  >
    <span>
      2020 Â© ARK.io | All rights reserved
    </span>
  </footer>
</div>
`;
